lib/api/truncate.js

```diff
@@ -0,0 +1,53 @@
+var SMB2Forge = require('../tools/smb2-forge');
+var SMB2Request = SMB2Forge.request;
+var BigInt = require('../tools/bigint');
+var FILE_WRITE_DATA = require('../structures/constants').FILE_WRITE_DATA;
+
+module.exports = function truncate(path, length, cb) {
+  var connection = this;
+  var file;
+  var fileLength = new BigInt(8, length);
+
+  function setFileSize(fileSizeSetted) {
+    SMB2Request(
+      'set_info',
+      {
+        FileId: file.FileId,
+        FileInfoClass: 'FileEndOfFileInformation',
+        Buffer: fileLength.toBuffer(),
+      },
+      connection,
+      function(err) {
+        if (err != null) {
+          cb && cb(err);
+          closeFile();
```

Comment:
The workflow is too complicated, always call the passed callback, and only this one.